lib/rubocop/cop/rails/find_each.rb

```diff
@@ -39,7 +39,7 @@ def autocorrect(node)
         private
 
         def method_chain(node)
-          [*node.ancestors, node].map(&:method_name)
+          [*node.descendants.take_while(&:send_type?), node].map(&:method_name)
```

Comment:
I would like to know why you choose `Enumerable#take_while`. For example, in the following cases, the behavior differs between `select` and` take_while`.

```ruby
User.order(:name).tap{ |foo| puts foo }.all.each { |u| u.something }
```

```
$ pry -rparser -rparser/ruby25 -rrubocop
[1] pry(main)> buffer = Parser::Source::Buffer.new('(string)')
[2] pry(main)> buffer.source = "User.order(:name).tap{ |foo| puts foo }.all.each { |u| u.something }"
[3] pry(main)> builder = RuboCop::AST::Builder.new
[4] pry(main)> parser = Parser::Ruby25.new(builder)
[5] pry(main)> node = parser.parse(buffer).send_node
[6] pry(main)> node.descendants.select(&:send_type?).map(&:method_name)
=> [:all, :tap, :order, :puts]
[7] pry(main)> node.descendants.take_while(&:send_type?).map(&:method_name)
=> [:all]
```

If use `take_while`, it cannot pick up all method chains, but if use `select`, it pick up the wrong `send` node in the block. It is difficult which is more desirable, but in terms of avoiding false positives, I guess `select` seems more desirable. WDYT?

Reply:
Yes, I think that both methods have false positives. For example, something like:

```ruby
User.where(id: Admin.select(:id)).all.each(&:something)
```

wouldn't register an offense because it will think that the `select` method for `Admin` is part of the method chain.

```
node.descendants.select(&:send_type?).map(&:method_name)
=> [:all, :where, :select]
node.descendants.take_while(&:send_type?).map(&:method_name).inspect
=> [:all, :where]
```

I really don't know which is more desirable.

Reply:
If we want to avoid false _positives_ then yes, it seems that `select` is safer. We can miss some real offenses but we won't get offenses that shouldn't be reported :+1: